### PR TITLE
53 known post db

### LINKED ISSUE #53: Create Known Posts Database Table
create a table that will hold the known posts and its associated comments.

### PR DESCRIPTION
**Pull request by weihanli101**
this closes #53

### CODE REVIEW

File: app/src/androidTest/java/ca/marshallasch/veil/database/DatabaseTest.java [hunk 1]
```diff
@@ -80,4 +82,10 @@ public void login()
         assertNull(database.login("email46@email.com", "abc123"));
     }
 
+    @Test
+    public void insertKnownPost(){
```
Comment (MarshallAsch) [inline app/src/androidTest/java/ca/marshallasch/veil/database/DatabaseTest.java]: You need way more tests then this, test all kinds of input and make sure it is what you expect it to be.

File: app/src/main/java/ca/marshallasch/veil/FragmentLogin.java [hunk 2]
```diff
@@ -71,12 +71,18 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
             db.close();
 
             // check that a user was found
-            if (user == null) {
+
+            //TODO: (DEBUGGING CODE) [START] RESTORE AFTER FINISH TESTING ===============================================
```
Comment (MarshallAsch) [inline app/src/main/java/ca/marshallasch/veil/FragmentLogin.java]: What on earth is this, you cant just turn off a block of code, if you want it to remember your login then add in a block to remember the username and password?

File: app/src/main/java/ca/marshallasch/veil/database/Database.java [hunk 3]
```diff
@@ -109,16 +108,16 @@ public void onCreate(SQLiteDatabase db)
...
         db.execSQL(NotificationContract.SQL_CREATE_POST_NOTIFICATION);
-        db.execSQL(KnownHashesContract.SQL_CREATE_KNOWN_HASHES);
         db.execSQL(UserContract.SQL_CREATE_USERS);
+        db.execSQL(KnownPostsContract.SQL_CREATE_KNOWN_HASHES);
 
     }
 
     @Override
     public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion)
     {
```
Comment (MarshallAsch) [inline app/src/main/java/ca/marshallasch/veil/database/Database.java]: leave in the old migration code, just add in the new migrations

File: app/src/main/java/ca/marshallasch/veil/database/Migrations.java [hunk 4]
```diff
@@ -15,7 +15,15 @@
      * In this version change the users table was added.
      * @param db the underlying database object
      */
-    static void v1ToV3(SQLiteDatabase db) {
+    /*static void upgradeV3(SQLiteDatabase db) {
```
Comment (MarshallAsch) [inline app/src/main/java/ca/marshallasch/veil/database/Migrations.java]: Leave in the old migration, just add in the new one

File: app/src/main/java/ca/marshallasch/veil/database/Migrations.java [hunk 5]
```diff
@@ -15,7 +15,15 @@
...
         db.execSQL(UserContract.SQL_CREATE_USERS);
+    }*/
+
+    /**
+     * In this version change the known posts table was added and the known hashes table was dropped
+     * @param db the underlying database object
+     */
+    static void upgradeV4(SQLiteDatabase db){
+        db.execSQL(KnownPostsContract.SQL_CREATE_KNOWN_HASHES);
```
Comment (MarshallAsch) [inline app/src/main/java/ca/marshallasch/veil/database/Migrations.java]: you removed the table but you did not remove it in the migration.

File: app/src/main/java/ca/marshallasch/veil/FragmentLogin.java (hunk 2 above)
Comment (weihanli101) [reply app/src/main/java/ca/marshallasch/veil/FragmentLogin.java]: Shoot I forgot to comment it back in.

Comment (weihanli101) [reply app/src/main/java/ca/marshallasch/veil/FragmentLogin.java]: resolved.

File: app/src/main/java/ca/marshallasch/veil/database/Database.java (hunk 3 above)
Comment (weihanli101) [reply app/src/main/java/ca/marshallasch/veil/database/Database.java]: Resolved

File: app/src/main/java/ca/marshallasch/veil/database/Migrations.java (hunk 4 above)
Comment (weihanli101) [reply app/src/main/java/ca/marshallasch/veil/database/Migrations.java]: resolved

File: app/src/main/java/ca/marshallasch/veil/database/Migrations.java (hunk 5 above)
Comment (weihanli101) [reply app/src/main/java/ca/marshallasch/veil/database/Migrations.java]: removed KnownHashes table in migration.

**Pull request merged**